src/common.h

```diff
@@ -140,15 +140,23 @@ struct v128 {
     static_assert(sizeof(T) <= sizeof(v), "Invalid cast!");
     assert((lane + 1) * sizeof(T) <= sizeof(v));
     T result;
+#if WABT_BIG_ENDIAN
```

Comment:
Refactor this into shared code?

Reply:
Shared with...?

Reply:
It seems that this pattern is common in many of the changes below (using the negative offset for `WABT_BIG_ENDIAN` on memcpy). Can you make a function that does this, `MemcpyEndianAware` or some other better name?

Reply:
So like a MemcpyEndianAwareAt that takes "source object", "destination object", "source size", "destination size", "source offset", "destination offset", and "copy size"?

(this is a lot of arguments... should it be `inline`? where should we put it?)

Reply:
`common.h` seems like the best choice (maybe near `Bitcast`, etc). It is a lot of arguments, but I think that will be OK.